pennylane/transforms/compile.py

```diff
@@ -162,6 +167,14 @@ def qfunc(x, y, z):
         ───RY(-1.57)─╰X─┤
 
     """
+    if expand_depth is not None:
+        warnings.warn(
+            "The expand_depth argument is deprecated and will be removed in version v0.40. ",
+            qml.PennyLaneDeprecationWarning,
+        )
+    else:
+        expand_depth = 5
```

Comment:
Since we're deprecating the behaviour of expansion depth, I'm wondering if it makes sense to just have the default be `None` rather than 5?

Reply:
Hmm, the way I see it is if a user is currently relying on the default `max_expansion=5` in `compile` (whether they're aware of it or not), changing the default to `None` could lead to different results depending on their circuit / implementation. 

Hence the `if` block to only raise a deprecation warning if the user tries to set something. 🤷🏼 

What do you think?

Reply:
Sounds good 😄